sql/2019/05_Third_Parties/05_04.sql

```diff
@@ -0,0 +1,20 @@
+#standardSQL
+# Percentage of total bytes that are from third party requests broken down by third party category by resource type.
```

Comment:
A little help for calculating the %:

```sql
#standardSQL
# Percentage of total bytes that are from third party requests broken down by third party category by resource type.
SELECT
  thirdPartyCategory,
  contentType,
  SUM(requestBytes) AS totalBytes,
  ROUND(SUM(requestBytes) * 100 / SUM(SUM(requestBytes)) OVER (), 2) AS pctBytes
FROM (
  SELECT
      resp_content_type AS contentType,
      respBodySize AS requestBytes,
      ThirdPartyTable.category as thirdPartyCategory
    FROM
      `httparchive.almanac.summary_requests`
    LEFT JOIN
      `lighthouse-infrastructure.third_party_web.2019_07_01` AS ThirdPartyTable
    ON NET.HOST(url) = ThirdPartyTable.domain
)
GROUP BY
  thirdPartyCategory,
  contentType
ORDER BY
  totalBytes DESC
```

I'm also seeing many similar content types, for example `text/javascript` vs `application/javascript`. The `type` field is more coarse and folds these together into a "script" type, if that's more helpful.

Reply:
thanks very much!!

re: text/javascript

I thought it cuts both ways and it's always easier to manually interpret groups if we have multiple, but given the leaning toward everything being done in the query I think that's a good change 👍